packages/webview_flutter/lib/platform_interface.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/widgets.dart';
+
+/// Interface for talking to the webview's platform implementation.
+///
+/// An instance implementing this interface is passed to the `onWebViewPlatformCreated` callback that is
+/// passed to [WebViewPlatformBuilder#onWebViewPlatformCreated].
+abstract class WebViewPlatform {
+  /// Loads the specified URL.
+  ///
+  /// If `headers` is not null and the URL is an HTTP URL, the key value paris in `headers` will
+  /// be added as key value pairs of HTTP headers for the request.
+  ///
+  /// `url` must not be null.
+  ///
+  /// Throws an ArgumentError if `url` is not a valid URL string.
+  Future<void> loadUrl(
+    String url,
+    Map<String, String> headers,
+  ) {
+    throw UnimplementedError(
+        "WebView loadUrl is not implemented on the current platform");
+  }
+
+  // As the PR currently focus about the wiring I've only moved loadUrl to the new way, so
+  // the discussion is more focused.
+  // In this temporary state WebViewController still uses a method channel directly for all other
+  // method calls so we need to expose the webview ID.
+  // TODO(amirh): remove this before publishing this package.
+  int get id;
+}
+
+typedef WebViewPlatformCreatedCallback = void Function(
+    WebViewPlatform webViewPlatform);
+
+/// Interface building a platform WebView implementation.
+///
+/// [WebView#platformBuilder] controls the builder that is used by [WebView].
+/// [AndroidWebViewPlatform] and [CupertinoWebViewPlatform] are the default implementations
+/// for Android and iOS respectively.
+abstract class WebViewBuilder {
+  /// Builds a new WebView.
```

Comment:
Nit: Should have an extra space above the dart doc?

Reply:
Should we? I wasn't aware that we enforce something like that, I can't find anything about it in the style guide, are we enforcing this convention?

Reply:
I asked because it seems other places in this file is doing that. If it is not enforced then I think we are ok. However, I do think it is better to keep the style consistent throughout a file or even a package.

Reply:
Not sure I understand, how can weI enforce that `"onWebViewPlatformCreated will be invoked after the platform specific [WebViewPlatform] implementation is created with the [WebViewPlatform] instance as a parameter."`